### ISSUE TITLE
addlocatorstrategy function arguments error

### ISSUE THREAD

**Issue by DMazona** [webdriverio/webdriverio#7111]
[//]: # NOTE: This repository only maintains packages that are listed in the [Readme](https://github.com/webdriverio/webdriverio/blob/main/README.md#packages). Please make sure that your issue is directly caused by one of these packages and if not file an issue in the correct 3rd party package repository.

**Environment (please complete the following information):**
 - **WebdriverIO version:** 7.7.4
 - **Mode:** standalone, i think
 - **If WDIO Testrunner, running sync/async:** async
 - **Node.js version:** 14.16.0
 - **NPM version:** 7.11.2
 - **Browser name and version:** chrome 91
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):** [e.g. @wdio/spec reporter, @wdio/selenium-standalone service]

**Config of WebdriverIO**
An example of how you configured WebdriverIO config

**Describe the bug**
I am in the process of evaluating the protractor conversion to webdriverio. I tried adding a custom locator strategy to mimic by.cssContainingText(locator, *regexp*) because this failed in conversion (expecting just a plain string). Following the guide for locators here https://webdriver.io/docs/selectors, it has two arguments for addLocatorStrategy function, (selector and root)
```
browser.addLocatorStrategy('myCustomStrategy', (selector, root) => {
    /**
     * scope should be document if called on browser object
     * and `root` if called on an element object
     */
    const scope = root ? root : document
    return scope.querySelectorAll(selector)
})
```
I tried doing something similar but vscode complains 

`Argument of type '(selector: string, root: HTMLElement) => NodeListOf<HTMLElement>' is not assignable to parameter of type '(selector: string) => CustomLocatorReturnValue'.ts(2345)`

It looks like it's expected the function argument to only contain one argument, instead of the two in the example code.

**To Reproduce**
copy example code from https://webdriver.io/docs/selectors#custom-selector-strategies
into working test script.

[Include code or an example repository that can easily be set up]

**Expected behavior**
I would expect the function to take multiple arguments. In this case, I wanted to pass in a regexp to search querySelectorAll using regexp, which I could not find an example of anywhere. ie Array.from(document.querySelectorAll(selector)).find(p=> p.textContent.match(regexp))

**Log**
If applicable, add logs to help explain your problem.
Create a [gist](https://gist.github.com) which is a paste of your _full_ Selenium/WebdriverIO logs, and link them here. Do _not_ paste your full Appium logs here, as it will make this issue very long and hard to read! If you are reporting a bug, _always_ include Selenium/WebdriverIO logs!

**Additional context**

> Please use the correct markdown

**Comment by christian-bromann** [comment]
Currently we only support a string selector but this should be an easy change (see [here](https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/utils/index.ts#L459)). I believe this is all we need to make you case possible. Would you mind taking a stab at it? Happy to support in the process.

**Comment by DMazona** [comment]
Maybe it wasn't clear, but my point in this bug is that the example code doesn't compile.

**Comment by christian-bromann** [comment]
Then we need to make sure our type definitions comply with our code examples.

**Comment by praveendvd** [comment]
@christian-bromann  @DMazona  it works fine for me , could you add a reproducible project as zip ? or a link to github project

**Comment by christian-bromann** [comment]
Fixed in #7226
